test/unit/ngsiv2/ngsiService/subscriptions-test.js

```diff
@@ -0,0 +1,367 @@
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {},
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        // As it can be seen in
+        // https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/user/walkthrough_apiv2.md#subscriptions,
+        // in NGSIv2, the `expires` element of the payload to create a subscription must be specified
+        // using the ISO 8601 standard format (e.g., 2016-04-05T14:00:00.00Z). However, in the IOTA,
+        // this value is load from the `deviceRegistrationDuration` property of the configuration file,
+        // which is expressed using ISO 8601 duration format (e.g., P1M). Therefore, in order to
+        // maintain compatibility with previous versions, for NGSIv2, the expires field is calculated
+        // adding the `deviceRegistrationDuration` property of the IOTA configuration file to the
+        // current date. This implies that in order to assert the value of the payload in the CB mock,
+        // we have to calculate dynamically the expected `expires` field.
+        // Please check line 86.
+        throttling: 'PT5S'
+    };
+
+describe('Subscription tests', function() {
+    beforeEach(function(done) {
+        var optionsProvision = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionMinimumDevice.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            // FIXME: change once NGISv2 device provisioning is implemented
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext',
+                    utils.readExampleFile('./test/unit/examples/contextRequests/createMinimumProvisionedDevice.json'))
+                .reply(200,
+                    utils.readExampleFile('./test/unit/examples/contextResponses/createProvisionedDeviceSuccess.json'));
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', function(body) {
+                    var expectedBody = utils.readExampleFile('./test/unit/ngsiv2/examples' +
+                        '/subscriptionRequests/simpleSubscriptionRequest.json');
+                    // Note that expired field ins not included in the json used by this mock as it is a dynamic
```

Comment:
Typo: ins -> is

Sorry, my fault, as author of the original text :$

Reply:
Fixed in cf32a5bf0a6de572d501273ce79a1a2fb76fd1cd